be/src/runtime/tablets_channel.cpp

```diff
@@ -105,11 +105,7 @@ void TabletsChannel::add_chunk(brpc::Controller* cntl, const PTabletWriterAddChu
     Sender& sender = _senders[request.sender_id()];
     std::lock_guard l(sender.lock);
 
```

Comment:
Yes, you're right, unlock `sender.lock` before calling `_load_channel->remove_tablets_channel(_index_id)` can also resolve the deadlock. 
Compared with unlock `sender.lock` before calling `LoadChannel::remove_tablets_channel()`, remove the acquirement of sender.lock in `LoadChannel::cancel()` totally can reduce the possibility of thread suspending.

Reply:
Since `LoadChannel::cancel()` is running in the bthread worker thread, blocking its running may affect other incoming requests serviced by brpc,  so I try to reduce the lock conflicts in  `LoadChannel::cancel()` by removing the acquirement of sender.lock.